python/lsst/consdb/hinfo_fix_missing.py

```diff
@@ -0,0 +1,258 @@
+                self.updates = {**self.updates, **attr(exposure_rec)}
+
+    def fix_telescope_position(self, exposure_rec: dict[str, Any]) -> dict[str, Any]:
+        """Attempts to fill in missing sky position columns.
+
+        It relies on s_ra and s_dec being present, and does
+        calculations to obtain altitude, azimuth and zenith
+        distance for the beginning, middle, and end of the
+        exposure as well as airmass.
+
+        It modifies the `exposure_rec` dictionary and returns nothing.
+
+        Parameters
+        ----------
+        exposure_rec : `dict[str, Any]`
+            The exposure record dictionary, (almost) ready to
+            be copied into consolidated database exposure
+            table.
+
+        Returns
+        -------
+        dict[str, Any]
+            Fixed columns in the original `exposure_rec` dictionary.
+        """
+
+        if exposure_rec["airmass"] is not None:
+            # No need to do calculations, it's already done.
+            return dict()
+
+        if exposure_rec["s_ra"] is None or exposure_rec["s_dec"] is None:
+            # Bail out because we don't have enough info.
+            return dict()
+
+        if exposure_rec["s_ra"] == 0.0 and exposure_rec["s_dec"] == 0.0:
+            # Bail out because ra and dec don't appear to be valid.
+            return dict()
+
+        # Convert from RA and Dec
+        s_ra, s_dec = map(lambda x: float(exposure_rec[x]), ("s_ra", "s_dec"))
+        location = EarthLocation.of_site("LSST")
+        obstimes = Time(
+            [
+                exposure_rec["obs_start_mjd"],
+                exposure_rec["exp_midpt_mjd"],
+                exposure_rec["obs_end_mjd"],
+            ],
+            format="mjd",
+            scale="tai",
+        )
+        coord = SkyCoord(s_ra, s_dec, unit="deg")
+        altaz = coord.transform_to(AltAz(obstime=obstimes, location=location))
+
+        # Get altaz calculations
+        altitude_start, altitude, altitude_end = altaz.alt.deg
+        azimuth_start, azimuth, azimuth_end = altaz.az.deg
+        zenith_distance_start, zenith_distance, zenith_distance_end = altaz.zen.deg
+
+        # Use K&Y 1989 model to compute airmass from altitude
+        airmass = None
+        if altitude >= 0 and altitude <= 90:
+            airmass = 1 / (sin(radians(altitude)) + 0.50572 * (altitude + 6.07995) ** -1.6364)
+
+        # Load them into the update dictionary.
+        update = {}
+        calculations = {
+            "altitude_start": altitude_start,
+            "altitude": altitude,
+            "altitude_end": altitude_end,
+            "azimuth_start": azimuth_start,
+            "azimuth": azimuth,
+            "azimuth_end": azimuth_end,
+            "zenith_distance_start": zenith_distance_start,
+            "zenith_distance": zenith_distance,
+            "zenith_distance_end": zenith_distance_end,
+            "airmass": airmass,
+        }
+        for k, v in calculations.items():
```

Comment:
While you're editing, can you expand these to useful names, or at least the full words?